src/services/inlayHints.ts

```diff
@@ -283,6 +291,15 @@ namespace ts.InlayHints {
 
                 addTypeHints(typeDisplayString, param.end);
             }
+
+            if (parameterNeedsParens) {
+                // ensure ")" is added after paremeter hints.
+                result.push({
+                    text: ")",
+                    position: node.parameters[node.parameters.length - 1].end,
+                    kind: InlayHintKind.Other,

```

Comment:
I think this still needs to be `Parameter` - but I'd need to consult with @mjbvz.